jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-api/src/main/java/org/jbpm/workbench/cm/util/CaseActionStatus.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.cm.util;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public enum CaseActionStatus {
+
+    READY("Ready"),
+    IN_PROGRESS("Reserved", "InProgress"),
+    COMPLETED_SUCCESS("Completed"),
+    COMPLETED_FAIL("Suspended", "Failed", "Error", "Exited", "Obsolete"),
+    ADHOC("N/A");
+
+    private final List<String> taskStatuses;
+
+    CaseActionStatus(final String... taskStatuses) {
+        this.taskStatuses = Collections.unmodifiableList(Arrays.asList(taskStatuses));
+    }
+
+    public static CaseActionStatus fromStatus(final String status) {
+        for (CaseActionStatus status1 : CaseActionStatus.values()) {
+            if (status1.getTaskStatuses().contains(status)) {
+                return status1;
+            }
+        }
+        return null;
+    }
+
```

Comment:
How about implementing fromStatus method in the same way it was implemented in CaseStageStatus enum?
Something along the lines of:
return Arrays.stream(CaseActionStatus.values()).filter(s -> s.getTaskStatuses().contains(status)).findFirst().get();

It's much more readable and in case the status argument turns out to be a bogus one, it's better from debugging point of view to have call to get() throw NoSuchElementException, then having switch statement in CaseActionItemView throw NullPointerException.